src/tree_graphviz.ml

```diff
@@ -57,7 +63,7 @@ open Tree_types
 (** Printing hint for conversion to graphviz. *)
 type dot_term =
   | DotDefa (* Default case *)
-  | DotAbst of tvar
+  | DotAbst of tvar * int
```

Comment:
Why do you need to keep a tvar argument here? Can't you remove it?

Reply:
It seems that if I don't keep the tvar, the substitution can't be done properly in `eval`. To see this, comment line 203 of `eval.ml` and run
```
dune exec -- lambdapi tests/OK/miller2.dk --verbose 3
```
to obtain the term
```
λx0 y0, pair 0 (p (d (λx0, ch x) x) (d (λx, sh y) x))
```

Reply:
Which line in the current version? Please give a permalink.

Reply:
https://github.com/gabrielhdt/lambdapi/blob/8d59ad2462bb54c1fca12fd0c05e6deb6f0b4173/src/eval.ml#L203

But I figured out what happens: during the tree build, the env_builder is built regarding the variables used. To wipe out the variables from the TC, I'd have to modify the tree build. I'll give it a try later on, but we might still want to merge what has been done here (for I think the code is already better).

Reply:
I prefer to get rid of tvar's completely first.